elytron/ELY-1686-legacy-properties-files-into-FileSystemRealm-utility.adoc

```diff
@@ -0,0 +1,109 @@
+= ELY-1686 Add utility to port legacy properties files into a FileSystemRealm
+:author:            Justin Michael Cook
+:email:             jucook@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+:issue-base-url:    https://issues.jboss.org/browse
+
+== Overview
+
+Currently, converting legacy security properties files into an Elytron FileSystemRealm
+must be done manually, there should be a migration tool that automates this conversion.
+This task is to add such a tool to WildFly Elytron Tool.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/ELY-1686[ELY-1686]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1093[EAP7-1093]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:msvehla@redhat.com[Martin Švehla]
+
+=== Affected Projects or Components
+
+* WildFly Elytron Tool
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* It should be possible to have legacy properties files converted into an Elytron
+FileSystemRealm automatically. This should be done by adding a command named
+FileSystemRealmCommand to the Elytron-Tool. This command will take in the name of
+the file or files to be converted as the only parameter. The command will parse the given
+legacy properties file to configure the new security domain, appropriate factories,
+appropriate subsystems, and http-interface if necessary. There should be two possible
+ways to specify the file(s): a singular file, or a comma-separated list of files.
+When specifying multiple files, the command should ensure each entry in the list is unique.
+
+** Singular file
+[source,bash]
+./bin/elytron-tool.sh filesystem-realm --file/-f <file>
+
+** List of files
+[source,bash]
+./bin/elytron-tool.sh filesystem-realm --file/-f <file1,file2,...>
```

Comment:
For determining the Elytron subsystem commands that can be used to configure things like a ```filesystem-realm``` and ```security-domain```, names for these resources will be required. These could be specified using parameters.

When attempting to convert multiple legacy properties file, multiple path parameters and multiple resource name parameters would be needed. To simplify things, perhaps you could make use of a descriptor file to do this, similar to what the Elytron Tool already does to bulk convert multiple vaults.